microsoft-edge/webview2/concepts/windowed-vs-visual-hosting.md

```diff
@@ -10,13 +10,14 @@ ms.date: 10/24/2022
 ---
 # Windowed vs. visual hosting of WebView2
 
-There are two options for hosting the Microsoft Edge WebView2 control in your app: windowed hosting and visual hosting.
+There are three options for hosting the Microsoft Edge WebView2 control in your app: windowed hosting, window to visual hosting, and visual hosting.
 
-If you use windowed hosting, which is a good starting point for most apps, you don't need to read this article.  If you want to provide a more custom user experience (UX), and want to use visual hosting, read this article.
+If you use windowed hosting, which is a good starting point for most apps, you don't need to read this article. If you want to provide a more custom user experience (UX) and want to use visual hosting, or if you are using windowed mode in specific scenarios and are experiencing persistent issues with DPI and scaling, read this article.
 
 | Approach | Description | Optimized for |
 |---|---|---|
 | Windowed hosting | The WebView2 control takes input from the operating system (OS).  The OS sends the input to the WebView2. | Displaying web content quickly and easily, without having to include features for inputs, outputs, and accessibility. |
+| Window to Visual hosting | A combination of windowed and visual modes. This mode provides a developer experience basically identical to windowed mode. The only developer from the app developer perspective is the need to set the `COREWEBVIEW2_FORCED_HOSTING_MODE` environment variable to `COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_VISUAL`. Like windowed mode, the OS sends input to the WebView2 and there is no need to use the visual hosting input APIs. But unlike windowed mode, the WebView2 output is displayed using a visual instead of a window as it is in visual hosting. This implementation detail allows us to prevent a number of DPI related bugs that pure windowed mode apps can run into. It is important to note that this hosting mode comes at the cost of losing pen input and handwriting support. | A developer experience nearly identical to windowed mode but with improved DPI/scaling handling and the caveat that pen input and handwriting is unsupported. |
```

Comment:
clarify "the WebView2 output is displayed using a visual instead of a window"

what's "a visual"? 
@johna-ms

Reply:
I will clarify language here. A "visual" is a basic graphical unit that can be used with other visuals to compose graphical experiences on Windows. There are a couple of Windows graphics APIs that expose this functionality namely dcomp and Windows::UI::Composition. The visual in visual hosting can be any one of IDCompositionVisual, IDCompositionTarget, or Windows::UI::Composition::Visual, which are visuals exposed through the aforementioned APIs. None of this is explained in the doc so I assumed it was fine to leave that abstracted away for a high level overview. Window to Visual uses the IDCompositionVisual specifically, but this is not exposed to or imo relevant to the developer experience.

Reply:
Actually I'm unsure how to clarify this without providing the above context. I think we could link to this article: https://learn.microsoft.com/en-us/windows/win32/directcomp/basic-concepts which is where I derive my explanation at least for dcomp, or we could leave as is with the understanding that a developer that knows what a visual in visual hosting is can assume the same for a visual in window to visual.

Reply:
result:
```
<!-- ------------------------------ -->
#### Window to Visual hosting: Similar experience as Windowed hosting with added benefits and tradeoff

_Window-to-Visual hosting_ means that the WebView2 content is displayed by using a visual that is hosted in an HWND (window), rather than hosting the content in a window directly or in a visual directly.

Window-to-Visual hosting is a combination of windowed and visual modes.

A _visual_ is a basic graphical unit that can be used with other visuals to compose graphical experiences on Windows. There are a couple of Windows graphics APIs that expose this functionality: `dcomp` and `Windows::UI::Composition`. The "visual" in "Visual hosting" can be any one of `IDCompositionVisual`, `IDCompositionTarget`, or `Windows::UI::Composition::Visual`, which are visuals that are exposed through the aforementioned APIs.  (Window to Visual hosting uses `IDCompositionVisual` specifically.)  See [Basic concepts](/windows/win32/directcomp/basic-concepts) in the Windows App Development > DirectComposition docs.
```
"a visual" occurs 5x in article; an odd-sounding usage, defining is warranted